src/webui/api/torrentscontroller.cpp

```diff
@@ -882,6 +882,35 @@ void TorrentsController::removeTrackersAction()
         torrent->forceReannounce();
 }
 
+void TorrentsController::removeTrackerAction()
+{
+    requireParams({u"urls"_s});
+
+    const QStringList urls = params()[u"urls"_s].split(u'|', Qt::SkipEmptyParts);
+
+    for (BitTorrent::Torrent *const torrent : asConst(BitTorrent::Session::instance()->torrents()))
+    {
+        QStringList trackersToRemove;
+        for (const BitTorrent::TrackerEntryStatus &trackerEntryStatus : asConst(torrent->trackers()))
+        {
+            for (const QString &urlStr : urls)
+            {
+                const QString url = QUrl::fromPercentEncoding(urlStr.toUtf8());
+                if ((trackerEntryStatus.url == url) || (QUrl(trackerEntryStatus.url).host() == url))
+                    trackersToRemove.append(trackerEntryStatus.url);
+            }
+        }
+
+        if (trackersToRemove.size() > 0)
+        {
+            torrent->removeTrackers(trackersToRemove);
+
+            if (!torrent->isStopped())
+                torrent->forceReannounce();
```

Comment:
FYI the GUI doesn't do this when removing a tracker via the sidebar, but the WebAPI does in `removeTrackersAction` directly above.

Reply:
This is going to be a bit off-topic, but still.
I don't really understand why when deleting a tracker all other trackers are force re-announced? Let's say there are 20 trackers, if I delete them one by one, the last one gets as many as 19 force announcements! FYI, some trackers will just ban you for a while for doing that. Force re-announce to all trackers also happens when editing one tracker, maybe in some other cases. Why is this done?

Reply:
> Why is this done?

I'm not really sure. It looks like I added this in 33351e3d8d442c02b7db346146bf7ab9283cdbd2 back in 2018, though the context for that decision has long since been forgotten. I called it out here as I suspected there would be opinions. If we can't identify a good reason for it, I'll remove it.

Reply:
I'm referring to this behavior of the client as a whole, not just web ui. IMO, it's not normal.

Reply:
@stalkerok

Some users might use default settings, had connection issues or have got trackers in same tier and many not contacted, so automatically reannouncing after any kind of editing can temporarily be good.

Some trackers list related issues have already mentioned before in a different PR. -> https://github.com/qbittorrent/qBittorrent/pull/19633#issuecomment-1742070441

Maybe no one hasn't made any issue reports yet, forgotten or just too few pay attention how things work.

Reply:
IIRC, libtorrent handle tracker changes by itself so you shouldn't do it manually. Moreover, `forceReannounce()` actually announces to all the trackers that isn't match the intention (and its name).

P.S. Existing code should be fixed as well but in separate PR.

P.P.S. I noticed this a long time ago and was going to fix it, but as always there is not enough time.

Reply:
I'll remove the reannounce behavior from this API and put up a new PR to remove the behavior from `removeTrackersAction`.

Reply:
https://github.com/qbittorrent/qBittorrent/pull/21077